scripts/check_doc_links.py

```diff
@@ -124,14 +124,35 @@ def _checks_all_html(file: Path, regex: Pattern = LINK_PATTERN_MD) -> None:
     Checks a file for matches to a pattern.
 
     :param file: the file path
-    :param all_files: all the doc file paths
     :param regex: the regex to check for in the file.
     """
     matches = regex.finditer(file.read_text())
     for _ in matches:
         raise ValueError("Markdown link found in file={}!".format(str(file)))
 
 
+def is_external_url(url: str) -> bool:
+    """
+    Check if an URL is an external URL.
+
+    :param url: the URL
+    :return: true if it is external, false otherwise.
+    """
+    return url.startswith("https") or url.startswith("http")
```

Comment:
suppose better to use:  `url.startswith("https://") or url.startswith("http://")` to avoid cases like https-forever.com